LoopKitUI/Views/Settings Editors/TherapySettingsViewModel.swift

```diff
@@ -10,23 +10,30 @@ import LoopKit
 import HealthKit
 
 public class TherapySettingsViewModel: ObservableObject {
-
-    private let initialTherapySettings: TherapySettings
+    public typealias SaveCompletion = (TherapySetting, TherapySettings) -> Void
+    
     @Published public var therapySettings: TherapySettings
     public var supportedInsulinModelSettings: SupportedInsulinModelSettings
-    public var didFinishStep: (() -> Void)?
+    private let didSave: SaveCompletion?
+
+    private let initialTherapySettings: TherapySettings
     let pumpSupportedIncrements: PumpSupportedIncrements?
+    let pumpSyncSchedule: PumpManager.SyncSchedule?
     let includeSupportSection: Bool
 
     public init(therapySettings: TherapySettings,
                 supportedInsulinModelSettings: SupportedInsulinModelSettings = SupportedInsulinModelSettings(fiaspModelEnabled: true, walshModelEnabled: true),
                 pumpSupportedIncrements: PumpSupportedIncrements? = nil,
-                includeSupportSection: Bool = true) {
+                pumpSyncSchedule: PumpManager.SyncSchedule? = nil,
+                includeSupportSection: Bool = true,
+                didSave: SaveCompletion? = nil) {
```

Comment:
Just curious—why make `didSave` optional?

Reply:
It's like a delegate function...it isn't required for the operation of TherapySettingsViewModel.